NeuroID/Utilities/NetworkMonitoringService.swift

```diff
@@ -26,21 +30,79 @@ internal class NetworkMonitoringService {
         monitor = NWPathMonitor()
     }
 
+    private func setupNoNetworkTask() {
+        noNetworkTask = DispatchWorkItem {
+            guard !(self.noNetworkTask.isCancelled) else {
+                return
+            }
+
+            // pause collection but don't flush events
+            NeuroID.pauseCollection(flushEventQueue: false)
+        }
+    }
+
+    private func setupResumeNetworkTask() {
+        resumeNetworkTask = DispatchWorkItem {
+            guard !(self.resumeNetworkTask.isCancelled) else {
+                return
+            }
+
+            NeuroID.resumeCollection()
+        }
+    }
+
     internal func startMonitoring() {
         monitor.start(queue: queue)
 
         monitor.pathUpdateHandler = { path in
-            self.isConnected = path.status == .satisfied
+            let connectionStatus = path.status == .satisfied
+
             self.getConnectionType(path)
 
             let nidEvent = NIDEvent(type: .networkState)
             nidEvent.iswifi = self.connectionType == .wifi
-            nidEvent.isconnected = self.isConnected
+            nidEvent.isconnected = connectionStatus
             nidEvent.attrs = [
-                Attrs(n: "connectionType", v: "\(self.connectionType.rawValue)")
+                Attrs(n: "connectionType", v: "\(self.connectionType.rawValue)"),
             ]
 
             NeuroID.saveEventToLocalDataStore(nidEvent)
+
+            if connectionStatus != self.isConnected {
+                self.isConnected = path.status == .satisfied
```

Comment:
Is `path.status == .satisfied` expected to change since it was assigned to `connectionStatus` on line 58?

Reply:
no, but that's a good call, I will use the connectionStatus var I created